### PR TITLE
Add support for Object Lock Compliance mode TC CEPH-83574057

### PR DESCRIPTION
**Pull request by TejasC88**
Steps performed:
- Add support for Governance mode in bucket level lock
- Add Legal Hold for all object uploads

Imp : Change in multisite_sync_policy.py is a missed line in last PR . It will be tested when Sync policy tests are added to cephCI

Pass log:
http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/object_lock_governance.txt

### CODE REVIEW

File: rgw/v2/tests/s3_swift/test_object_lock.py [hunk 1]
```diff
@@ -77,13 +76,27 @@ def test_exec(config, ssh_con):
...
+                    s3_conn_client.put_object_lock_configuration(
+                        Bucket=bucket_name_to_create,
+                        ObjectLockConfiguration={
+                            "ObjectLockEnabled": "Enabled",
+                            "Rule": {
+                                "DefaultRetention": {"Mode": "GOVERNANCE", "Days": 1}
+                            },
+                        },
+                    )
+
```
Comment (viduship) [inline rgw/v2/tests/s3_swift/test_object_lock.py:99]: @TejasC88 , Can we define a function for the put_object_lock call in the resuable and call it here?

Comment (TejasC88) [reply rgw/v2/tests/s3_swift/test_object_lock.py:99]: @viduship updated ,pls review

**Review by ckulal — COMMENTED** [review]
Looks good

**Review by anrao19 — APPROVED** [review]

**Review by viduship — APPROVED** [review]
looks good

**Pull request merged**